product_contract_variable_quantity/models/product_template.py

```diff
@@ -5,26 +5,36 @@
 
 
 class ProductTemplate(models.Model):
-    _inherit = 'product.template'
+    _inherit = "product.template"
 
     qty_type = fields.Selection(
         selection=[
-            ('fixed', 'Fixed quantity'),
-            ('variable', 'Variable quantity'),
+            ("fixed", "Fixed quantity"),
+            ("variable", "Variable quantity"),
         ],
         required=False,
-        default='fixed',
+        default="fixed",
         string="Qty. type",
+        compute="_compute_qty_type",
+        store=True,
+        readonly=False,
     )
     qty_formula_id = fields.Many2one(
-        comodel_name="contract.line.qty.formula", string="Qty. formula"
+        comodel_name="contract.line.qty.formula",
+        string="Qty. formula",
+        compute="_compute_qty_formula_id",
+        store=True,
+        readonly=False,
     )
 
-    @api.onchange('is_contract')
-    def _change_is_contract(self):
-        """ Clear the relation to contract_template_id when downgrading
-        product from contract
-        """
-        if not self.is_contract:
-            self.qty_type = False
-            self.qty_formula_id = False
+    @api.depends("is_contract")
+    def _compute_qty_type(self):
+        for rec in self:
+            if not rec.is_contract:
+                rec.qty_type = False
```

Comment:
Suggested change: `self.filtered(lambda rec: not rec.is_contract).qty_type = False`